craft_parts/executor/part_handler.py

```diff
@@ -173,6 +173,11 @@ def _run_overlay(self, step_info: StepInfo) -> StepState:
                 ) as ctx:
                     ctx.install_packages(overlay_packages)
 
+            # remove temporary files and package cache from layer
+            shutil.rmtree(self._part.part_layer_dir / "var/cache", ignore_errors=True)
+            shutil.rmtree(self._part.part_layer_dir / "var/tmp", ignore_errors=True)
+            shutil.rmtree(self._part.part_layer_dir / "tmp", ignore_errors=True)
+
```

Comment:
this should not be done by craft-parts but by the user instead. This can break layering.

Reply:
I don't think this can break layering, we're removing temporary files directly from the topmost layer and exposing content that previous layers might have placed there. Since this is performed for all layers, these directories are never part of any layer and the base layer contents are always visible. That said, we could leave this cleanup operation to the user or to the application, but it's something that should be performed at some point before packing the primed payload.

Reply:
Filesystem cleanup calls removed.